wcfsetup/install/files/lib/system/template/TemplateEngine.class.php

```diff
@@ -1045,14 +1045,15 @@ protected function getCompileFilePrefix(string $templateName): string
     private function getSharedTemplateGroupID(): int
     {
         if (!isset($this->sharedTemplateGroupID)) {
-            $sql = "SELECT  templateGroupID
-                    FROM    wcf1_template_group
-                    WHERE   templateGroupFolderName = ?";
-            $statement = WCF::getDB()->prepare($sql);
-            $statement->execute(['_wcf_shared/']);
-
-            $this->sharedTemplateGroupID = $statement->fetchSingleColumn();
+            /** @var TemplateGroup $templateGroup */
```

Comment:
I much prefer an `\array_find()` combined with a check that will throw a `RuntimeException` if the result is `null`. Right now your code will read from an uninitialized property if there is no match.